### PR TITLE
feat: use official jdtls wrapper script

### PR DESCRIPTION
**Pull request by agorgl**
Implements #1819

### CODE REVIEW

**Comment by mjlbach** [comment]
@mfussenegger do you know of any cons other than the python 3.9 dependency of using the wrapper script?

**Comment by agorgl** [comment]
Fixed minor lint warning

**Comment by agorgl** [comment]
Having written manually the wrappers for launching correctly jdtls for my old vim configuration in all systems (lin,win,osx) in the past, I think using the upstream solution is the way to go.

The upstream wrapper script separates the usage between the read-only shared config passed with the flag `osgi.sharedConfiguration.area` and the read-write user config passed with the `-configuration` flag by us.

**Comment by agorgl** [comment]
Added also support for the `--jvm-arg` flag that the jdtls wrapper script accepts. Custom jvm arguments can be passed with the `JDTLS_JVM_ARGS` environment variable as a space separated list of arguments, that will be converted to multiple `--jvm-arg=<param>` args when passed to the jdtls script. This will allow for example tweaking the jvm arguments or integration with external tools like lombok:
```
export JDTLS_JVM_ARGS="-javaagent:$HOME/.local/share/java/lombok.jar -Xbootclasspath/a:$HOME/.local/share/java/lombok.jar"
```

File: lua/lspconfig/server_configurations/jdtls.lua [hunk 1]
```diff
@@ -151,19 +135,6 @@ you can keep reading here.
 For manual installation you can download precompiled binaries from the
 [official downloads site](http://download.eclipse.org/jdtls/snapshots/?d)
```
Comment (schrieveslaach) [inline lua/lspconfig/server_configurations/jdtls.lua:136]:
```suggestion
[official downloads site](http://download.eclipse.org/jdtls/snapshots/?d)
and ensure that the `PATH` variable contains the `bin` directory of the extracted archive. 
```

Comment (agorgl) [reply lua/lspconfig/server_configurations/jdtls.lua:136]: Fixed!

**Comment by mfussenegger** [comment]
> @mfussenegger do you know of any cons other than the python 3.9 dependency of using the wrapper script?

No real cons as far as I am aware off.

Just keep in mind that the changes here will probably break the config for all people who've been using it so far.

**Comment by agorgl** [comment]
Any updates on this?

**Comment by mjlbach** [comment]
I have a slight concern about introducing a python 3.9 dependency, although debian bullseye has 3.9 by default, that's the main hold-up.

**Comment by agorgl** [comment]
I think that any user/dev having the whole jdk in his system will probably have python too.
Also as you said almost all mainstream distros almost always come bundled with python!

**Comment by mfussenegger** [comment]
Might want to hold off until it can properly parse the java version: https://github.com/eclipse/eclipse.jdt.ls/issues/2060

**Comment by agorgl** [comment]
Interesting, i'm using the latest version and it seems to be working fine

**Comment by BojanStipic** [comment]
@agorgl thank you very much for this PR, I am looking forward to seeing this merged.

> Added also support for the `--jvm-arg` flag that the jdtls wrapper script accepts. Custom jvm arguments can be passed with the `JDTLS_JVM_ARGS` environment variable as a space separated list of arguments, that will be converted to multiple `--jvm-arg=<param>` args when passed to the jdtls script. This will allow for example tweaking the jvm arguments or integration with external tools like lombok:
> 
> ```
> export JDTLS_JVM_ARGS="-javaagent:$HOME/.local/share/java/lombok.jar -Xbootclasspath/a:$HOME/.local/share/java/lombok.jar"
> ```

I think it would be very useful to also include this in the documentation, to ease the setup process for newcomers as Lombok is commonly used. There is currently no way to find this out, other than digging into the source code.

**Comment by schrieveslaach** [comment]
@BojanStipic, @agorgl the option `-Xbootclasspath` might not work always. On my system lombok stopped working when I used it. The [project's documentation](https://projectlombok.org/setup/ecj) states that older versions of Lombok or Java requires that option:

>  You may have to add the following VM argument, if you're using an older version of Lombok or Java: 

I assume that Java 11 does not requires it in general and jdt.ls require at least Java 11. 

So I think `-Xbootclasspath` shouldn't documented as a default option.

**Comment by agorgl** [comment]
Added documentation about JDTLS_JVM_ARGS environment variable as per the above suggestions!

**Comment by agorgl** [comment]
@mfussenegger My fix for jdtls java version parsing landed in master: https://github.com/eclipse/eclipse.jdt.ls/pull/2063
I think we are good to go!

**Comment by agorgl** [comment]
@mfussenegger java version parsing fix is now included on latest stable builds, available both on AUR (Arch Linux) and scoop (Windows), maybe we should merge this?

**Comment by mfussenegger** [comment]
I don't maintain nvim-lspconfig, not up to me.

**Comment by agorgl** [comment]
@mjlbach Is there anything I can do to help merge this?

**Comment by agorgl** [comment]
Fixed stylua linting error in latest force push

**Comment by agorgl** [comment]
Ping @justinmk as per his request!

**Pull request merged**